gear/identifer_lookup/src/python/identifer_app/run.py

```diff
@@ -0,0 +1,131 @@
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+def get_identifiers(rds_parameters: RDSParameters,
+                    adcid: int) -> Dict[str, Identifier]:
+    """Gets all of the Identifier objects from the identifier database using
+    the RDSParameters.
+
+    Args:
+      rds_parameters: the credentials for RDS MySQL with identifiers database
+      adcid: the center ID
+    Returns:
+      the dictionary mapping from PTID to Identifier object
+    """
+    identifiers = {}
+    identifers_session = create_session(rds_parameters)
+    with identifers_session as session:
+        identifiers_repo = IdentifierRepository(session)
+        center_identifiers = identifiers_repo.list(adc_id=adcid)
+        if not center_identifiers:
+            # pylint: disable=(not-an-iterable)
+            identifiers = {
+                identifier.ptid: identifier
+                for identifier in center_identifiers
+            }
+
+    return identifiers
+
+
+def get_adcid(proxy: FlywheelProxy, file_id: str) -> Optional[int]:
+    """Get the adcid of the center group that owns the file.
+
+    Args:
+      proxy: the flwheel proxy object
+      file_id: the ID for the file
+    Returns:
+      the ADCID for the center
+    """
+    file = proxy.get_file(file_id)
+    group = file.parents.group
+    center = CenterGroup(group=group, proxy=proxy)
+    return center.center_id()
+
+
+# pylint: disable=(too-many-locals)
+def main():
+    """The Identifiers Lookup gear reads a CSV file with rows for participants
+    at a single ADRC, and having a PTID for the participant. The gear looks up
+    the corresponding NACCID, and creates a new CSV file with the same
+    contents, but with a new column for NACCID.
+
+    Writes errors to a CSV file compatible with Flywheel error UI.
+    """
+
+    with GearToolkitContext() as gear_context:
+        gear_context.init_logging()
+
+        try:
+            parameter_store = ParameterStore.create_from_environment()
+            dry_run = gear_context.config.get("dry_run", False)
+            proxy = FlywheelProxy(client=Client(parameter_store.get_api_key()),
+                                  dry_run=dry_run)
+            rds_parameters = parameter_store.get_rds_parameters(
+                param_path=get_config(gear_context=gear_context,
+                                      key='rds_parameter_path'))
+        except ConfigParseError as error:
+            log.error('Incomplete configuration: %s', error.message)
+            sys.exit(1)
+        except ParameterError as error:
+            log.error('Parameter error: %s', error)
+            sys.exit(1)
+
+        file_input = gear_context.get_input('input_file')
+        if not file_input:
+            log.error('Missing input file')
+            sys.exit(1)
+
+        file_id = file_input['object']['file_id']
+        adcid = get_adcid(proxy=proxy, file_id=file_id)
+        if not adcid:
```

Comment:
Can adcid be 0? We use that for the sample center. I think this would evaluate to True in that case

Reply:
Yes. Replaced with explicit check for None